File: Python3.x/191-Number of 1 Bits.py
class Solution(object):
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        res = 0
        for i in str(bin(n)[2:]):
            if i == '1': res += 1
        return res


class Solution(object):
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        return bin(n).count('1')


class Solution(object):
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        res = 0
        while n:
            n = n & (n - 1)  # Think of a number in binary n = XXXXXX1000, n - 1 is XXXXXX0111. n & (n - 1) will be XXXXXX0000 which is just cancel the last 1
            res += 1
        return res


class Solution(object):
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        res = 0
        while n:
            n, res = n & (n - 1), res + 1
        return res

File: C++/191-Number_of_1_Bits.cpp
class Solution {
public:
    int hammingWeight(uint32_t n) { // uint32_t是unsigned int
        int res = 0;
        while (n) {
            n = n & (n - 1);
            res++;
        }
        return res;
    }
};


class Solution {
public:
    int hammingWeight(uint32_t n) {
        return n == 0 ? 0 : 1 + hammingWeight(n & (n - 1));
    }
};
